pkg/util/set.go

```diff
@@ -32,6 +33,18 @@ func NewStringSet(items ...string) StringSet {
 	return ss
 }
 
+// KeySet creates a StringSet from a keys of a map[string](? extends interface{}).  Since you can't describe that map type in the Go type system
+// the reflected value is required.
+func KeySet(theMap reflect.Value) StringSet {
```

Comment:
I'd never tried to do it on a generic map type... can't really think of anything better, but ick...